lib/consul/async/json_endpoint.rb

```diff
@@ -184,6 +184,7 @@ def _handle_error(http)
 
       def fetch
         options = {
+          tls: { verify_peer: conf.tls_verify_peer },
```

Comment:
FAR: it would have been better to have a common base class for all endpoint types.